persistence/sql/migrations/sql/20230405000000000001_create_session_token_exchanges.mysql.up.sql

```diff
@@ -0,0 +1,15 @@
+CREATE TABLE session_token_exchanges (
+    id CHAR(36) NOT NULL PRIMARY KEY,
+    nid CHAR(36) NOT NULL,
+    flow_id CHAR(36) NOT NULL,
+    session_id CHAR(36) DEFAULT NULL,
+    code VARCHAR(64) NOT NULL,
+
+    created_at timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    updated_at timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP
+);
+
+-- Relevant query:
+--   SELECT * from session_token_exchanges
+--   WHERE flow_id = ? AND nid = ? AND code = ? AND session_id IS NOT NULL AND code <> '';
```

Comment:
I need to revisit the indices because we now need to lookup an exchange by (nid, code) as well (without the flow), but in any case I think the combination of nid and flow_id should already uniquely identify a single exchange, so adding more columns to the index does not help with the lookup speed. Correct?

Reply:
I now added two indices:
* (flow_id, nid) for queries specifying the flow_id
* (code, nid) for queries specifying the code

Both combinations should return at maximum one query, and flow_id and code respectively are more selective than nid.